compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -5685,8 +5685,9 @@ private BErrorType createErrorType(BType lhsType, BType rhsType, BType detailTyp
         BErrorType lhsErrorType = (BErrorType) lhsType;
         BErrorType rhsErrorType = (BErrorType) rhsType;
 
-        BErrorType errorType = createErrorType(detailType, lhsType.flags, env);
-        errorType.tsymbol.flags |= rhsType.flags;
+        long flags = lhsType.flags | rhsType.flags | Flags.PUBLIC;  // Anonymous error intersection types must be public
+
```

Comment:
```suggestion

```

Reply:
And also there are few places which have extra lines. Better to remove those as well.